upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1422,6 +1424,41 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 		}
 	}
 
+	if cluster.Spec.GetCloudProvider() == kops.CloudProviderOpenstack {
+		keyset := keysets["kops-controller-bootstrap"]
+		if !ig.HasAPIServer() {
+			if keyset != nil && keyset.Primary != nil && keyset.Primary.PrivateKey != nil {
+				b, err := keyset.Primary.PrivateKey.AsBytes()
+				if err != nil {
+					return nil, nil, fmt.Errorf("unable to find privatekey: %w", err)
+				}
+
+				key, err := jwt.ParseRSAPrivateKeyFromPEM(b)
+				if err != nil {
+					return nil, nil, fmt.Errorf("parsing pem key failed: %w", err)
+				}
+				now := time.Now().UTC()
+				claims := jwt.MapClaims{
+					"ig":  ig.Name,
+					"iat": now.Unix(),
+					"nbf": now.Unix(),
+					"exp": now.Add(time.Hour).Unix(),
+				}
+				token, err := jwt.NewWithClaims(jwt.SigningMethodRS256, claims).SignedString(key)
+				if err != nil {
+					return nil, nil, fmt.Errorf("sign token failed: %w", err)
+				}
+				bootConfig.BootstrapToken = token
+			}
+		} else {
+			kopsControllerPublicKeys, err := keyset.ToPublicKeys()
+			if err != nil {
+				return nil, nil, fmt.Errorf("encoding kops-controller-bootstrap keys: %w", err)
+			}
+			bootConfig.BootstrapPublicKeys = kopsControllerPublicKeys
```

Comment:
This code is not necessary.

Reply:
Actually, here it needs to call `loadCertificates()` with `includeKeypairID` of `false`.